handler/handler_test.go

```diff
@@ -76,6 +75,16 @@ func (m *MockMetricStore) Ready() error {
 	return nil
 }
 
+func ctxWithParams(params map[string]string, mainReq *http.Request) context.Context {
+	ctx := mainReq.Context()
+
+	for key, value := range params {
+		ctx = context.WithValue(ctx, key, value)
```

Comment:
Isn't it more appropriate to use `route.WithParam` for this?

Perhaps that also fixes the problem you are seeing in the tests.